src/validation/__tests__/NoSchemaIntrospectionCustomRule-test.js

```diff
@@ -0,0 +1,153 @@
+// @flow strict
+
+import { describe, it } from 'mocha';
+
+import {
+  GraphQLSchema,
+  GraphQLObjectType,
+  GraphQLString,
+  __EnumValue,
+} from '../../type/index';
+
+import { NoSchemaIntrospectionCustomRule } from '../rules/custom/NoSchemaIntrospectionCustomRule';
+
+import { expectValidationErrorsWithSchema } from './harness';
+
+function expectErrors(queryStr) {
+  return expectValidationErrorsWithSchema(
+    schema,
+    NoSchemaIntrospectionCustomRule,
+    queryStr,
+  );
+}
+
+function expectValid(queryStr) {
+  expectErrors(queryStr).to.deep.equal([]);
+}
+
+const schema = new GraphQLSchema({
+  query: new GraphQLObjectType({
+    name: 'Query',
+    fields: {
+      someQuery: {
+        type: new GraphQLObjectType({
+          name: 'SomeType',
+          fields: {
+            someField: { type: GraphQLString },
+            introspectionField: { type: __EnumValue },
+          },
+        }),
+      },
+    },
+  }),
+});
```

Comment:
@danielrearden I merged #2608 so now `buildSchema` should work fine can you please use it instead.